bagitobjecttransfer/recordtransfer/models.py

```diff
@@ -39,138 +42,364 @@ def get_full_name(self):
 
 
 class UploadSession(models.Model):
-    """Represents a file upload session, that may or may not be split into
-    multiple parallel uploads
+    """Represents a file upload session. This model is used to track the files that a
+    user uploads during a session.
     """
 
+    class SessionStatus(models.TextChoices):
+        """The status of the session."""
+
+        CREATED = "CR", _("Session Created")
+        EXPIRED = "EX", _("Session Expired")
+        UPLOADING = "UG", _("Uploading Files")
+        COPYING_IN_PROGRESS = "CP", _("File Copy in Progress")
+        STORED = "SD", _("All Files in Permanent Storage")
+        COPYING_FAILED = "FD", _("Copying Failed")
+        REMOVING_IN_PROGRESS = "RP", _("File Removal in Progress")
+        DELETED = "DL", _("All Files Removed")
+
     token = models.CharField(max_length=32)
     started_at = models.DateTimeField()
+    status = models.CharField(
+        max_length=2, choices=SessionStatus.choices, default=SessionStatus.CREATED
+    )
 
     @classmethod
-    def new_session(cls):
-        """Start a new upload session"""
+    def new_session(cls) -> "UploadSession":
+        """Start a new upload session."""
         return cls(token=get_random_string(length=32), started_at=timezone.now())
 
     @property
-    def upload_size(self):
-        """Get total size (in bytes) of all uploaded files in this session"""
-        return sum(f.file_upload.size for f in self.get_existing_file_set())
+    def upload_size(self) -> int:
+        """Get total size (in bytes) of all uploaded files in this session. This includes the size
+        of both temporary and permanent files.
+        """
+        if self.status in (
+            self.SessionStatus.EXPIRED,
+            self.SessionStatus.DELETED,
+        ):
+            raise ValueError(
+                f"""Cannot get upload size from session {self.token} because the session has
+                expired or has been deleted"""
+            )
```

Comment:
Using a triple """ preserves the extra whitespace on the indented line (starting with "expired") and preserves the newline after the word "has". This should be changed to a string on a single line:

```suggestion
            raise ValueError(
                f"Cannot get upload size from session {self.token} because the session has "
                "expired or has been deleted"
            )
```

It's not as clean looking but doesn't have that unwanted side effect that """ does.

Reply:
Addressed by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/318/commits/27a5f765eb73a3faa444e2a40977b38728ce2c2d